platform/ios/Integration Tests/MGLTabBarControllerTests.m

```diff
@@ -0,0 +1,74 @@
+
+#import "MGLMapViewIntegrationTest.h"
+
+@interface MGLTabBarControllerTests : MGLMapViewIntegrationTest
+
+@property (nonatomic) UIViewController *viewController;
+@property (nonatomic) UITabBarController *tabController;
+@end
+
+@implementation MGLTabBarControllerTests
+
+- (void)setUp {
+    [super setUp];
+    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"IntegrationTest" bundle:[NSBundle mainBundle]];
+    [self.window setRootViewController:(UITabBarController *)[storyboard instantiateViewControllerWithIdentifier:@"TabController"]];
+    
+    // Access the tab controller from the story board, then access one of the childViewControllers.
+    self.tabController = (UITabBarController *)self.window.rootViewController;
+    self.viewController = self.tabController.childViewControllers[0];
+    
+    [self.viewController.view addSubview:self.mapView];
+    
+    MGLPointAnnotation *annot = [[MGLPointAnnotation alloc] init];
+    annot.coordinate = self.mapView.centerCoordinate;
+    [self.mapView addAnnotation:annot];
+}
+
+- (void)testViewControllerNil {
+    XCTAssertNotNil(self.viewController);
+}
+
+- (void)testSwitchingTabs {
+    
+    XCTestExpectation *expectation = [self expectationWithDescription:@"Able to switch the tabs repeatedly without lag"];
+    __block NSInteger counter = 0;
+    
+    __block NSTimer *repeatingTimer = [NSTimer scheduledTimerWithTimeInterval:0.1 repeats:YES block:^(NSTimer * _Nonnull timer) {
+        
+        counter++;
+        NSLog(@"%li", counter);
+        self.tabController.selectedIndex = counter % 2;
+        if (counter > 41) {
+            
+            [repeatingTimer invalidate];
+            repeatingTimer = nil;
+            
+            MGLMapCamera *camera = [MGLMapCamera cameraLookingAtCenterCoordinate:CLLocationCoordinate2DMake(10, 10) altitude:100 pitch:0 heading:0];
+            [self.mapView flyToCamera:camera withDuration:4 completionHandler:^{
+                [expectation fulfill];
+            }];
+            
+            [self measureBlock:^{
```

Comment:
We shouldn't need to call `updateFromDisplayLink` the `waitForExpectations:..` below should handle that IIRC.

I hadn't considered using `measureBlock` - that's an interesting idea - but I don't think it's appropriate here.